### PR TITLE
Fix voice announcement rounding problem

### PR DESCRIPTION
**Pull request by WPettersson**
We extracted integral parts before rounding, but String.format does
rounding which caused occasional issues. This fixes that by rounding all
values as we fetch them, so the String.format rounding won't break
anything.

Fixes: #1285

### CODE REVIEW

File: src/main/java/de/dennisguse/opentracks/services/announcement/VoiceAnnouncementUtils.java [hunk 1]
```diff
@@ -61,7 +61,9 @@ static Spannable getAnnouncement(Context context, TrackStatistics trackStatistic
                 throw new RuntimeException("Not implemented");
         }
 
-        double distanceInUnit = distance.toKM_Miles(unitSystem);
+        // Total distance is announced to two decimal places, so round early to
+        // avoid rounding issues (see #1285)
+        double distanceInUnit = Math.round(distance.toKM_Miles(unitSystem) * 100) / 100.0;
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/services/announcement/VoiceAnnouncementUtils.java:66]: extract into a method?

**Review by dennisguse — COMMENTED** [review]
Adding a test to VoiceAnnouncementUtilsTest would be great.

And why not use BigDecimal?

File: src/main/java/de/dennisguse/opentracks/services/announcement/VoiceAnnouncementUtils.java (hunk 1 above)
Comment (WPettersson) [reply src/main/java/de/dennisguse/opentracks/services/announcement/VoiceAnnouncementUtils.java:66]: I was thinking about adjusting `appendDecimalUnit` to instead take a double, and a precision, which seemed to make sense.

**Comment by WPettersson** [comment]
> Adding a test to VoiceAnnouncementUtilsTest would be great.

Think I can manage that, probably not today though.

> And why not use BigDecimal?

Even with BigDecimal, we need to do rounding before splitting into integer and fractional parts. I had a BigDecimal version worked out, but realised that using BigDecimal is unnecessary as I doubt we'll overflow a double.

**Comment by WPettersson** [comment]
Pushed a new commit which puts the rounding code inside `appendDecimalUnit` and also has some tests that pass and should be checking for this issue in the future.

**Comment by dennisguse** [comment]
Looks good.
Same question: why not use big decimal?

**Comment by WPettersson** [comment]
Even with BigDecimal, we need to do rounding before splitting into integer and fractional parts. I had a BigDecimal version worked out, but realised that using BigDecimal is unnecessary as I doubt we'll overflow a double.

**Pull request merged**